monocular_pose_estimator/package.xml

```diff
@@ -9,27 +9,23 @@
 
   <buildtool_depend>catkin</buildtool_depend>
 
-  <build_depend>roscpp</build_depend>
-  <build_depend>cv_bridge</build_depend>
-  <build_depend>image_transport</build_depend>
-  <build_depend>dynamic_reconfigure</build_depend>
-  <build_depend>sensor_msgs</build_depend>
-  <build_depend>geometry_msgs</build_depend>
-  <build_depend>monocular_pose_estimator_lib</build_depend>
-  <build_depend>cmake_modules</build_depend>
-  <build_depend>nodelet</build_depend>
-  <build_depend>camera_calibration_parsers</build_depend>
-
-  <run_depend>roscpp</run_depend>
-  <run_depend>cv_bridge</run_depend>
-  <run_depend>image_transport</run_depend>
-  <run_depend>dynamic_reconfigure</run_depend>
-  <run_depend>sensor_msgs</run_depend>
-  <run_depend>geometry_msgs</run_depend>
-  <run_depend>monocular_pose_estimator_lib</run_depend>
-  <run_depend>nodelet</run_depend>
-  <run_depend>opencv</run_depend>
-  <run_depend>v4l-utils</run_depend>
+  <build_depend>yaml-cpp</build_depend>
+  <build_export_depend>yaml-cpp</build_export_depend>
```

Comment:
in the CMakeLists.txt you link against `yaml-cpp_LIBRARIES`, so, most likely this is also an exec_dependency